### PR TITLE
Add tests for ingredient model

### LINKED ISSUE #83: Add tests for the ingredient model
to ensure it works as expected.

### PR DESCRIPTION
**Pull request by Diwu-Yi**

### CODE REVIEW

**Review by WM71811 — DISMISSED** [review]
LGTM

File: src/test/java/seedu/address/model/ingredient/AmountTest.java [hunk 1]
```diff
@@ -1,14 +1,49 @@
...
+    }
+
+    @Test
+    public void constructor_invalidAmount_throwsIllegalArgumentException() {
+        String invalidAmount = "";
+        assertThrows(IllegalArgumentException.class, () -> new Amount(invalidAmount));
+    }
+
     @Test
     void isValidAmount() {
```
Comment (ureshiiYing) [inline src/test/java/seedu/address/model/ingredient/AmountTest.java]: Missing public attribute? Applies for all the other tests.

File: src/test/java/seedu/address/model/ingredient/UniqueIngredientListTest.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+    @Test
+    public void setIngredient_targetIngredientNotInList_throwsIllegalArgumentException() {
+        assertThrows(IllegalArgumentException.class, () -> uniqueIngredientList
+                .setIngredient(new Ingredient(new IngredientName("Olong Tea")), OOLONG_TEA_VALID_AMOUNT));
+    }
+
+    @Test
+    public void setIngredient_editedIngredientIsSameIngredient_throwsNoChangeIngredientException() {
+
```
Comment (ureshiiYing) [inline src/test/java/seedu/address/model/ingredient/UniqueIngredientListTest.java:56]: Unnecessary white space?

**Review by ureshiiYing — COMMENTED** [review]
Perhaps you should add the public attribute to the test cases. Other than that LGTM.

File: src/test/java/seedu/address/model/ingredient/AmountTest.java (hunk 1 above)
Comment (Diwu-Yi) [reply src/test/java/seedu/address/model/ingredient/AmountTest.java]: yep, will add the access modifier.

File: src/test/java/seedu/address/model/ingredient/UniqueIngredientListTest.java (hunk 2 above)
Comment (Diwu-Yi) [reply src/test/java/seedu/address/model/ingredient/UniqueIngredientListTest.java:56]: yep, will detele.

**Review by Persdre — APPROVED** [review]
LGTM

**Review by yanbingtao — APPROVED** [review]
LGTM

**Pull request merged**